src/gt4py/storage/utils.py

```diff
@@ -87,7 +88,7 @@ def normalize_default_origin(default_origin, mask=None):
     if sum(mask) < len(default_origin):
         new_default_origin = [h for i, h in enumerate(new_default_origin) if mask[i]]
 
-    return tuple(new_default_origin)
+    return Index(new_default_origin)
```

Comment:
Changing the return type here and in the above `normalize_shape` could affect their functionality in the storages (haven't checked).

Reply:
Good catch, these have been reverted.

Reply:
I actually switched these back to returning Index and Shape, since I feel somewhat strongly that they should be such. It simplifies some of the downstream logic. Please let us know if you think this breaks anything, but so far we haven't seen a test failure due to it.

Reply:
ok, so I changed the type on the storage side instead